build_test.go

```diff
@@ -224,193 +220,6 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 		Expect(buffer.String()).To(ContainSubstring("Configuring launch environment"))
 	})
 
-	context("when the build plan entry does not include a version", func() {
-		it.Before(func() {
-			entryResolver.ResolveCall.Returns.BuildpackPlanEntry = packit.BuildpackPlanEntry{
-				Name: "mri",
-				Metadata: map[string]interface{}{
-					"launch": true,
-				},
-			}
-		})
-
-		it("picks the newest version", func() {
-			result, err := build(packit.BuildContext{
-				CNBPath: cnbDir,
-				Stack:   "some-stack",
-				Plan: packit.BuildpackPlan{
-					Entries: []packit.BuildpackPlanEntry{
-						{
-							Name: "mri",
-							Metadata: map[string]interface{}{
-								"launch": true,
-							},
-						},
-					},
-				},
-				Layers: packit.Layers{Path: layersDir},
-			})
-			Expect(err).NotTo(HaveOccurred())
-			Expect(result).To(Equal(packit.BuildResult{
-				Layers: []packit.Layer{
-					{
-						Name: "mri",
-						Path: filepath.Join(layersDir, "mri"),
-						SharedEnv: packit.Environment{
-							"GEM_PATH.override": "/some/mri/gems/path",
-						},
-						BuildEnv:         packit.Environment{},
-						LaunchEnv:        packit.Environment{},
-						ProcessLaunchEnv: map[string]packit.Environment{},
-						Build:            false,
-						Launch:           true,
-						Cache:            false,
-						Metadata: map[string]interface{}{
-							mri.DepKey: "",
-						},
-					},
-				},
-				Launch: packit.LaunchMetadata{
-					BOM: []packit.BOMEntry{
-						{
-							Name: "mri",
-							Metadata: paketosbom.BOMMetadata{
-								Version: "mri-dependency-version",
-								Checksum: paketosbom.BOMChecksum{
-									Algorithm: paketosbom.SHA256,
-									Hash:      "mri-dependency-sha",
-								},
-								URI: "mri-dependency-uri",
-							},
-						},
-					},
-				},
-			}))
-		})
-	})
-
-	context("when the build plan entry version source is from $BP_MRI_VERSION", func() {
```

Comment:
@robdimsdale why does this test need to fully go?

Reply:
Both of these tests (no-version, and BP_MRI_VERSION) are pretty weak tests in my opinion. All we test for - and all that we are able to test for - is that whatever version string is returned from one mock (`entryResolver`) is provided to the next mock (`dependencyManager`). There is no other use for these versions; there is no other logic that interacts with them in `build.go`. We validate that we invoke `dependencyManager` with the string returned from `entryResolver` in the first, default, test, and so we do not need any other tests.

To phrase this another wya - we could write any number of tests with any number of random strings, and all we could test for is that whatever string `entryResolver` returns is provided to `dependencyManager`. There is an infinite number of possible tests, but only one thing to test for.

If we choose to remove the `entryResolver` interface, and call `packit/draft.Planner` direction (like we did in [cpython](https://github.com/paketo-buildpacks/cpython/pull/327)) then these tests would become slightly more meaningful, as then we would be validating real behavior for the resolution of entries rather than a mock. But, even then, there would still be limited value because `build.go` would not actually do anything with the version beyond providing it to the mocked `dependencyManager`.

This is a bit nuanced, so please let me know if this doesn't make sense.